skimage/segmentation/_felzenszwalb_cy.pyx

```diff
@@ -34,16 +35,22 @@ def _felzenszwalb_cython(image, double scale=1, sigma=0.8,
         Larger sigma gives smother segment boundaries.
     min_size : int, optional (default 20)
         Minimum component size. Enforced using postprocessing.
+    multichannel : bool, optional (default: True)
+        Whether the last axis of the image is to be interpreted as multiple
+        channels. A value of False, for a 3D image, is not currently supported.
 
     Returns
     -------
     segment_mask : (N, M) ndarray
         Integer mask indicating segment labels.
     """
-    if image.ndim != 3:
-        raise ValueError("This algorithm works only on single or " 
-                         "multi-channel 2d images. "
-                         "Got image of shape %s" % str(image.shape))
+    if multichannel and image.shape[2] not in [1, 3]:
+        warn(RuntimeWarning("Got image with third dimension of %s. This image "
+                            "will be interpreted as a multichannel 2d image, "
+                            "which may not be intended." % str(image.shape[2])))
+    elif not multichannel:
+        raise ValueError("This algorithm works only on single or "
+                         "multi-channel 2d images. ")
```

Comment:
This seems a bit weird... So `multichannel = False` is a `ValueError` every time here, since we are always only passing 3D images from `felzenszwalb`... I suggest removing the `multichannel` parameter altogether for the private function, and simply assuming that the final dimension is a channel, and warning when the number of channels exceeds 3.